llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -519,22 +521,27 @@ class ConfigStrategy : public PostPipelinerStrategy {
 private:
   const std::string Name;
   std::set<int> SuccSiblingScheduled;
+  std::set<int> PredSiblingScheduled;
   std::function<bool(const SUnit &A, const SUnit &B)> Discriminators[Size] = {
       [&](const SUnit &A, const SUnit &B) { return false; },
-      [&](const SUnit &A, const SUnit &B) { return A.NodeNum < B.NodeNum; },
+      [&](const SUnit &A, const SUnit &B) {
+        return TopDown ? A.NodeNum < B.NodeNum : A.NodeNum > B.NodeNum;
+      },
       [&](const SUnit &A, const SUnit &B) {
```

Comment:
Nit: The name of that priority function is now a bit confusing, it's only `Latest` for the top-down mode.